### PR TITLE
Resolution to QR code scanning issue

### PR DESCRIPTION
**Pull request by VibroAxe**
Adjusts the javascript handling to fix the tab change mechanic as described in #114 
Resolves absolute pathing issues from #111 
General quality of life codes fixes within QR code

### CODE REVIEW

**Comment by VibroAxe** [comment]
This probably wants a squash commit

File: octoprint_SpoolManager/static/js/SpoolManager.js [hunk 1]
```diff
@@ -723,28 +723,32 @@ $(function() {
...
+            var tabHashCode = window.location.hash;
+            //we can only contain -spoolId on the very first page
+            if (tabHashCode.includes("#tab_plugin_SpoolManager-spoolId")){
+                var selectedSpoolId = tabHashCode.replace("-spoolId", "").replace("#tab_plugin_SpoolManager", "");
+                console.info('Loading spool: '+selectedSpoolId);
+                // - Load SpoolItem from Backend
+                // - Open SpoolItem
+                self.apiClient.callSelectSpool(selectedSpoolId, function(responseData){
+                    //Select the SpoolManager tab
+                    $('a[href="#tab_plugin_SpoolManager"]').tab('show')
```
Comment (VibroAxe) [inline octoprint_SpoolManager/static/js/SpoolManager.js:740]: @OllisGit Do you know if there Is there a better way of doing this?

**Comment by VibroAxe** [comment]
Also fixes #111

**Comment by VibroAxe** [comment]
Added some general quality of life improvements (mainly url_for), this might want to be a squash/refactored commit

**Comment by OllisGit** [comment]
Thanks a lot for the PR!

My fix looks similar!.

**Comment by VibroAxe** [comment]
@OllisGit Any chance of a version bump to pull this into 1.3.4? It's working off master but not of 1.3.3

**Pull request merged**